jhu-idc/idc-isle-dc
# Theming

Prefer "Squash and Merge" as Git history in this branch is not important. The UI work is done in a different Github repo, this branch was used by the UI devs to share a Drupal config and snapshots. 

* Adds two Composer repos
  * `jhu-idc/idc-ui-theme` : the theme
  * `jhu-idc/idc-ui-module` : Drupal module that defines some blocks/components to be used by the theme
* Configuration changes (non-exhaustive list)
  * Install custom theme and set as default
  * Custom front page
  * Custom global contact page - includes installing and light configuration of a Captcha module and an associated Image Captcha module
  * Custom taxonomy pages
  * Set global header & footer
  
## Testing

We need to make sure that the theme has been included without disrupting other functionality in IDC.

* Check presence of front page (https://islandora-idc.traefik.me/)
* Check Contact page (https://islandora-idc.traefik.me/contact)
  * Captcha should appear if you are NOT logged in. It should not appear if you are logged in
  * I don't know if email is functional, but this page is configured to send mail to `webmaster@localhost.com` - this is set in Structure > Contact Forms > Website feedback > Edit

### Known issues

Some issues with the included pages:

* Top menu with links to other libraries has a max width. This is not intended, this menu is supposed to extend across the screen
* Inline admin actions tabs (such as View, Edit, etc) have white text color, making it difficult to see
* Contact page should have a picture of a very nice looking building or something. For now we just have a placeholder space
* ~~Item pages don't show info at the moment, will be addressed soon™~~

Closes https://github.com/jhu-idc/iDC-general/issues/274
